### ISSUE TITLE
Process Instance Model is not working with Accept application/json

### ISSUE THREAD

**Issue by mauriziovitale** [Activiti/Activiti#1873]
When I call the API 
http://{{domain}}/myapp-rb/v1/process-instances/4159a15f-3f1b-11e8-aa83-0a586460031e/model
with 
Accept: application/json 

Current result:
I'm getting a 406

Expecting result:
The JSON as we have for the process-definitions API
http://{{domain}}/myapp-rb/v1/process-definitions/4159a15f-3f1b-11e8-aa83-0a586460031e/model

**Comment by salaboy** [comment]
@mauriziovitale can you please elaborate a little bit more about the content that you are expecting. 
For example.. you cannot do 
```GET http://{{domain}}/myapp-rb/v1/process-definitions/4159a15f-3f1b-11e8-aa83-0a586460031e/model```
because that ID is not a process definition id is a process instance ID.. so that call will never work.

Doing a GET to:
```http://{{domain}}/myapp-rb/v1/process-instances/4159a15f-3f1b-11e8-aa83-0a586460031e/model```
produces a image/svg content which is decorated with the current activities that are running.

Is it ok if we just return the json content of the model as in process definition? meaning with no decoration? Is it a good first step? how do you use the JSON content? Why we can't use the SVG instead?

**Comment by salaboy** [comment]
@ryandawsonuk @erdemedeiros @igdianov @daisuke-yoshimoto this highlight the need of moving away from Runtime Bundle the generation of the process diagrams. This should be responsibility of the Query Service. User Interfaces or Client Applications will rely on the Runtime Bundle to be alive to render process diagrams and that is a not a good approach. It is adding "read" responsibility to the RB which we should avoid as much as possible.

**Comment by salaboy** [comment]
@mauriziovitale let's have a quick catch up on this one.. the more a look into the problem that we are trying to solve here the more I think that we need to return the decorated SVG and the Front End should use that, not matter which is the current/previous solution. We need to agree on a way moving forward that doesn't hurt us in the future even if it means to modify some of the existing components.

**Comment by salaboy** [comment]
This PR adds the endpoint that @mauriziovitale https://github.com/Activiti/activiti-cloud-runtime-bundle-service/pull/195 but we shouldn't merge until we agree on what is the best way to deal with diagrams

**Comment by daisuke-yoshimoto** [comment]
@salaboy @igdianov 

In the meantime, if it only corresponds to the content of the issue, is it acceptable for [salaboy's PR](https://github.com/Activiti/activiti-cloud-runtime-bundle-service/pull/195) content? How about moving the process diagram as a separate issue?

Also, I think it is certainly a duty of Query Service too, but is it hard to change it so that process diagrams can be generated with Query Service?
* Does Query Service manage bpmn's resources? In order to generate a process diagram, it is necessary to access the source bpmn.
* We also need to add third party librarys for generating diagram to Query Service.

In addition, is it currently a major way to generate svg of the process diagram on the server side? I think it is common to generate a process diagram from bpmn on the front end. I think it is common to generate a process diagram from bpmn on the front end using librarys like bpmn-io.

**Comment by daisuke-yoshimoto** [comment]
@salaboy 
> but if BPMN.io already provides some tooling .. we should use that..
> I would assume that BPMN.io will handle json or xml content to render the diagrams.. so maybe we don't even need to care about SVG.

By using bpmn-io's bpmn-js, we can display the process diagram from bpmn source.
We can also export it as svg.
(ex)
https://rawgit.com/bpmn-io/bpmn-js-examples/master/starter/viewer.html
https://github.com/bpmn-io/bpmn-js-examples/blob/master/starter/viewer.html

Also, it is better to generate a process diagram at the front end using bpmn-js for the following two reasons.
1. It is more resource efficient than generating it at the back end. It seems that the processing time will also be shortened.
2. We can create a rich process diagram from the process diagram generated by activiti-image-generator. We can use the same visual process diagram as when modeling.

**Comment by salaboy** [comment]
@daisuke-yoshimoto this is good stuff.. what do we need to have in query then? does BPMN.io uses a json representation of the process? or just the XML?

**Comment by daisuke-yoshimoto** [comment]
@salaboy 

I think that it is both.
We can display process diagrams from XML.
Even if there is only json, we can convert json to bpmn by using bpmn-moddle library
https://github.com/bpmn-io/bpmn-moddle
    
> what do we need to have in query then?
   
I think that we need to provide the endpoint to get executions.
Developers can generate process diagram by using bpmn-js with bpmn and execution information.

**Comment by salaboy** [comment]
@mauriziovitale @erdemedeiros as we discussed we will not merge this and we will prioritise having the BPMN model in query for consumption on the front end using the BPMN.io renderer
